godot-ffi/src/global.rs

```diff
@@ -184,28 +200,21 @@ pub enum GlobalLockError<'a, T> {
 enum InitState<T> {
     Initialized(T),
     Pending(fn() -> T),
-    TransientInitializing,
     Failed,
 }
 
 impl<T> InitState<T> {
-    fn unwrap_ref(&self) -> &T {
+    fn initialized(&self) -> Option<&T> {
         match self {
-            InitState::Initialized(t) => t,
-            _ => {
-                // SAFETY: This method is only called from a guard, which can only be obtained in Initialized state.
-                unsafe { std::hint::unreachable_unchecked() }
-            }
+            InitState::Initialized(t) => Some(t),
+            _ => None,
         }
     }
```

Comment:
Nice change, as it moves the entire unsafety to the caller and doesn't make assumptions about where it's called from :+1: 

Name also makes sense, maybe `as_initialized[_mut]` would be even clearer, to indicate the reference return?